model/src/main/java/org/projectnessie/model/Contents.java

```diff
@@ -30,22 +30,52 @@
 @Schema(
     type = SchemaType.OBJECT,
     title = "Contents",
-    oneOf = {IcebergTable.class, DeltaLakeTable.class, SqlView.class},
+    oneOf = {
+      IcebergTable.class,
+      IcebergSnapshot.class,
```

Comment:
why have separate API objects to track the global vs local state? I see how we would want that distinction inside the version store but to me this is leaking version store implemention details to the user.

Reply:
`Put` operations contain the per-reference state - the iceberg-snapshot-id.
`PutGlobal` operations contain the global-state (iceberg-table-metadata) to set and the expected global-state

A single `Put` would otherwise leak global-state "logic" into per-reference-state-only types (deltalake + sql-views)

There are two options:
* `Put` w/ per-reference state + `PutGlobal` w/ global-state + expected-state or
* `Put` w/ a composite object and expected global-state. IIRC we discarded this option of a `Put` "with everything" a couple weeks ago.

Reply:
I had hoped that `IcebergTable` would be `{Long snapshotId, String metadataLocation}` and the REST API or the version store would split that into a global change and a local change. When performing a get we then jsut return the iceberg table. To me the user shouldn't see global state at all and there should be one iceberg related Contents. IMHO global state is purely an artifact of the kernel and not somethign a user should care about.

Reply:
That was actually the initial approach (adding `snapshotId` to `IcebergTable`).
Anyway, I've updated this PR accordingly. It's rather simple now (see PR description).